aws/data_source_aws_dx_private_virtual_interface.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsDxPrivateVirtualInterface() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsDxPrivateVirtualInterfaceRead,
+
+		Schema: map[string]*schema.Schema{
+			"virtual_interface_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"connection_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vpn_gateway_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dx_gateway_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vlan": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"bgp_asn": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"bgp_auth_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"address_family": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"customer_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"amazon_address": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"mtu": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"jumbo_frame_capable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"tags": tagsSchemaComputed(),
+		},
+	}
+}
+
+func dataSourceAwsDxPrivateVirtualInterfaceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	vif, err := dxVirtualInterfaceRead(d.Get("virtual_interface_id").(string), conn)
+	if err != nil {
+		return err
+	}
+	if vif == nil {
+		log.Printf("[WARN] Direct Connect virtual interface (%s) not found", d.Get("virtual_interface_id").(string))
```

Comment:
Seems like this should be returning this as an error to the operator 😄

Reply:
Fixed